enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-provider-snippet.ts

```diff
@@ -1,14 +1,28 @@
 export const ANALYTICS_PROVIDER_SNIPPET_MINIMAL = `
 import {createContext, useState} from 'react'
 
-// Used for the example theme switcher component
-export const AnalyticsContext = createContext({})
+/**
+ * @typedef {Object} AnalyticsContextType
+ * @property {'light'|'dark'} themeKey - The current theme key.
+ * @property {null} [email] - Email of the user.
+ * @property {(themeKey: 'light'|'dark') => void} setThemeKey - Function to update the theme.
+ */
+
+/**
+ * @type {React.Context<AnalyticsContextType>}
+ */
+export const AnalyticsContext = createContext(
+  /** @type {AnalyticsContextType} */ ({})
+);
 
 // Demo provider that adds the state for the example theme switcher component.
 // Delete this once you've played around with the theme switcher, and use your
 // own application's theming instead.
 export const AnalyticsProvider = ({children}) => {
-  const [themeKey, setThemeKey] = useState('light')
+  /**
+   * @type {['light'|'dark', React.Dispatch<React.SetStateAction<'light'|'dark'>>]}
+   */
+  const [themeKey, setThemeKey] = useState(/** @type {'light'|'dark'} */ ('light'));
```

Comment:
I think the `/** @type {'light'|'dark'} */` in the initial value is redundant as we already typed it above in `type {['light'|'dark', React.Dispatch<React.SetStateAction<'light'|'dark'>>]}`. We can rely on inference here to not clutter up the snippets too much.